### PR TITLE
Bump network proxy to v0.0.14

### PR DESCRIPTION
**Pull request by Jefftree**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Bump network proxy image to v0.0.14. Includes some logging changes and keepalive fix (https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/154)

See full list of changes [here](https://github.com/kubernetes-sigs/apiserver-network-proxy/compare/v0.0.12...v0.0.13)

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -1876,6 +1875,7 @@ function prepare-konnectivity-server-manifest {
   params+=("--agent-service-account=konnectivity-agent")
   params+=("--kubeconfig=/etc/srv/kubernetes/konnectivity-server/kubeconfig")
   params+=("--authentication-audience=system:konnectivity-server")
+  params+=("--v=3")
```
Comment (BenTheElder) [inline cluster/gce/gci/configure-helper.sh:1878]: is this related?
can we get a better commit message?

Comment (Jefftree) [reply cluster/gce/gci/configure-helper.sh:1878]: sorry this is not intended to be included in the PR. There were some issues with the tests that I'm trying to debug. Marking this as WIP.

**Comment by Jefftree** [comment]
/hold 
The network proxy tests are acting a bit strange, going to do some more debugging.

**Comment by spiffxp** [comment]
possible github issue, please link if you run into this on other PR's https://github.com/kubernetes/test-infra/issues/19910

**Comment by Jefftree** [comment]
This PR is dependent on https://github.com/kubernetes/k8s.io/pull/1409

**Comment by cheftako** [comment]
/retest
kubernetes/k8s.io#1409 has merged.

**Comment by Jefftree** [comment]
```
$ git fetch https://github.com/kubernetes/kubernetes.git pull/96389/head
From https://github.com/kubernetes/kubernetes
 * branch                    refs/pull/96389/head -> FETCH_HEAD
$ git merge --no-ff e99facea8932bab962aa49e4071d2b6c73ef3411
merge: e99facea8932bab962aa49e4071d2b6c73ef3411 - not something we can merge
# Error: exit status 1
```

I'll try rebasing

**Comment by cheftako** [comment]
Guessing its kubernetes/test-infra#19910 as spiffxp indicated.

**Comment by Jefftree** [comment]
Seems to be working sporadically for certain PRs

/retest

**Comment by Jefftree** [comment]
Tests are passing now

/assign @lavalamp

File: go.mod [hunk 2]
```diff
@@ -526,7 +526,7 @@ replace (
 	modernc.org/strutil => modernc.org/strutil v1.0.0
 	modernc.org/xc => modernc.org/xc v1.0.0
 	rsc.io/pdf => rsc.io/pdf v0.1.1
-	sigs.k8s.io/apiserver-network-proxy/konnectivity-client => sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.12
+	sigs.k8s.io/apiserver-network-proxy/konnectivity-client => sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.14
```
Comment (lavalamp) [inline go.mod:529]: Are there client/server incompatibilities between .12 and .14?

Comment (Jefftree) [reply go.mod:529]: All clients should be backwards compatible, so .12 client will work with a .14 image as well.
